core/mm2Property.js

```diff
@@ -531,25 +531,33 @@ GLOBAL.Property = {
       view: 'foam.ui.FunctionView',
       help: 'Function for validating property value.',
       preSet: function(_, f) {
-        var str = f.toString();
-        var deps = str.
-          match(/^function[ _$\w]*\(([ ,\w]*)/)[1].
-          split(',').
-          map(function(name) { return name.trim(); });
+        if ( ! f.dependencies ) {
+          var str = f.toString();
+          var deps = str.match(/^function[ _$\w]*\(([ ,\w]*)/)[1];
+          if ( deps )
+            deps = deps.split(',').map(function(name) { return name.trim(); });
+          else
+            deps = [];
 
-        var f2 = function() {
-          var args = [];
-          for ( var i = 0 ; i < deps.length ; i++ )
-            args.push(this[deps[i]]);
-          return f.apply(this, args);
-        };
+          var f2 = function() {
+            var args = [];
+            for ( var i = 0 ; i < deps.length ; i++ )
+              args.push(this[deps[i]]);
+            return f.apply(this, args);
+          };
 
-        f2.dependencies = deps;
-        f2.toString = function() { return f.toString(); };
+          f2.dependencies = deps;
+          f2.toString = function() { return f.toString(); };
 
-        return f2;
+          return f2;
+        } else {
```

Comment:
No need for else if the end of the if case is return.

I prefer thes tyle 

if ( a ) {
  return a;
}
return b;